nifi-commons/nifi-record/src/main/java/org/apache/nifi/serialization/record/ResultSetRecordSet.java

```diff
@@ -239,6 +239,11 @@ private DataType getDataType(final int sqlType, final ResultSet rs, final int co
                     // Default scale is used to preserve decimals in such case.
                     decimalScale = resultSetScale > 0 ? resultSetScale : defaultScale;
                 }
+                // Scale can be bigger than precision in some cases (Oracle, e.g.) If this is the case, assume precision refers to the number of
+                // decimal digits and thus precision = scale
+                if (decimalScale > decimalPrecision) {
+                    decimalPrecision = decimalScale;
```

Comment:
It would be useful to add a unit test method in `ResultSetRecordSetTest` to verify the expected behavior.